src/Helpers/selectAllCheckboxHelper.test.js

```diff
@@ -0,0 +1,85 @@
+import selectAllCheckbox from './selectAllCheckboxHelper';
+import messages from '../Messages';
+import { intl } from '../Utilities/IntlProvider';
+
+const mockParameters = {
+        selectedItems: [],
+        selectorHandler: jest.fn(),
+        items: { 
+            data: [{ id: 'a' },{ id: 'b' }], 
+            meta: { total_items: 10 } 
+        },
+        fetchResource: jest.fn().mockReturnValue({ 
+                            payload: Promise.resolve({
+                                data: [{ id: 'testID' }]
+                            })
+                        }),
+        multiRow: true
+};
+
+describe('SelecAllCheckboxHelper', () => {
+    it('Should selectAllCheckbox return valid object for bulk select', () => {
+        const { items } = selectAllCheckbox(mockParameters);
+        expect(items).toEqual([
+            {
+                title: intl.formatMessage(messages.selectNone),
+                onClick: expect.any(Function)
+            },
+            {
+                title: `${intl.formatMessage(messages.selectPage, { count: 1 })}`,
+                onClick: expect.any(Function)
+            },
+            {
+                title: `${intl.formatMessage(messages.selectAll, { count: 10 })}`,
+                onClick: expect.any(Function)
+            }
+        ])
+    });
+
+    it('Should handleOnCheckboxChange call handleSelectAll', () => {
+        const { handleOnCheckboxChange } = selectAllCheckbox(mockParameters);
+        handleOnCheckboxChange();
+        
+        expect(mockParameters.fetchResource).toHaveBeenCalledWith({ page_size: 10, page: 1 });
+    });
+
+    it('Should handleOnCheckboxChange call handleUnselectAll', () => {
+        const { handleOnCheckboxChange } = selectAllCheckbox({ ...mockParameters, selectedItems: ['a']});
+        handleOnCheckboxChange();
+    });
+
+    it('Should select a page', () => {
+        const { items } = selectAllCheckbox(mockParameters);
+        items[1].onClick();
+        expect(mockParameters.selectorHandler).toHaveBeenCalledWith(true,['a', 'b']);
+    });
+
+    it('Should set multirow to default false safely', () => {
+        let testParams = mockParameters;
+        delete testParams.multiRow;
+        selectAllCheckbox(testParams);
```

Comment:
what do you expect here after calling `selectAllCheckbox` ?

Reply:
Honestly, I could not find any way to test `multiRow` inside `selectCheckboxHelper`, as it not accessible. It has only one variable  `selectPageNumber` associated to, but it is also not accessible. 

Thus, I thought it is enough, if we do not get any errors, if `multiRow` is not provided